[PL-52014] Update delegate docs on number of parallel executions (#7194)
harness/developer-hub: docs/platform/delegates/delegate-concepts/delegate-requirements.md

* The delegate is installed in your network and connects to the Harness Manager.

 :::danger delegate resources

>  Remember that the memory and CPU requirements are for the delegate only. Your delegate host/pod/container will need more computing resources for its operations systems and other services, such as Docker or Kubernetes.

Author: ```suggestion
Memory and CPU requirements are for the delegate only. Your delegate host/pod/container requires additional computing resources for its operating system and other services, such as Docker or Kubernetes.
```

> The resource requirements of the delegate container vary depending on the type of tasks or executions. For example, CI-only delegates can handle hundreds of parallel pipelines, whereas, for CD Terraform tasks, a single task might require a 2Gi container.

Author: ```suggestion
The resource requirements for the delegate container depend on the type of tasks or executions. For instance, CI-only delegates can handle hundreds of parallel pipelines. However, for CD Terraform tasks, a single task might require a 2Gi container due to Terraform's memory requirements. Each Terraform command needs at least 500MB of memory.
```

>  The resource requirements of the delegate container vary depending on the type of tasks or executions. For instance, CI-only delegates can handle hundreds of parallel pipelines, whereas for CD Terraform tasks, a single task might require a 2Gi container.

Author: ```suggestion
The resource requirements of the delegate container vary depending on the type of tasks or executions. For example, CI-only delegates can handle hundreds of parallel pipelines, whereas, for CD Terraform tasks, a single task might require a 2Gi container.
```

 :::

One delegate size does not fit all use cases, so Harness lets you pick from several options:

| Replicas | Required memory / CPU | Maximum parallel deployments and builds across replicas |
| :--: |  :--: | :--: |
| 1 | 2 GB / 0.5 CPU | 10 |
| 2 | 4 GB / 1 CPU | 20 |
| 4 | 8 GB / 2 CPU | 40 |

> | 8 | 16 GB / 4 CPU | 80 |

Reviewer: I think we used to have this in our old installation flows as options, but those don't apply any more. I think we should just remove this.

> One delegate size does not fit all use cases, so Harness lets you pick from several options:
> 
> | Replicas | Required memory / CPU | Maximum parallel deployments and builds across replicas |
> | :--: |  :--: | :--: |
> | 1 | 2 GB / 0.5 CPU | 10 |
> | 2 | 4 GB / 1 CPU | 20 |
> | 4 | 8 GB / 2 CPU | 40 |
> | 8 | 16 GB / 4 CPU | 80 |

Author: ```suggestion
```
